dashboard/app/controllers/api/v1/sections_students_controller.rb

```diff
@@ -88,6 +90,8 @@ def student_params
       :name,
       :sharing_disabled,
       :password,
+      :secret_picture,
+      :secret_words,
```

Comment:
These look unused - do we need to add these as permitted params if we actually just want to expose the reset operation?

Reply:
i think these are still needed because `secret_picture` and `secret_words` will be part of `student_params` when `@student.update(student_params)` is called (since the `reset_secrets` method doesn't save/update the user)

Reply:
But... how do the newly-generated secrets get into `params` to be filtered here?  As I understand it, `reset_secrets` updates these values on the `@student` instance so they should be persisted next time that instance is `save`d, which is probably happening as a side effect of the update.  I think that is totally independent of this bit of code though, which is operating purely on the `params` object and knows nothing about the model instance.

There might be a huge gap in my understanding of Rails + permitting params here though.

Reply:
@madelynkasula I'm not sure I follow. Can you explain how this API will be used? Presumably the client will never supply `secret_picture` and `secret_words` values, but rather pass along `secrets=reset_secrets` to trigger the reset. Therefore these are not needed, and should not be permitted here in the `student_params`.

The `@student` object will be committed to the db, with the reset secrets, in [update](https://github.com/code-dot-org/code-dot-org/pull/22724/files#diff-650471959dae2ecd050f72a9fe41739bR28) on line 28.

Reply:
ohhh, i see. i misunderstood how strong params would handle this case and thought that any param being updated on `@student` needed to be part of strong params or the save would fail. re-running my tests without those lines confirms that isn't true! i'll update

Reply:
The strong params are only relevant where called. It's convention to contain the call in a method, like here in `student_params`, which you can see it called explicitly in the update on line 28
```ruby
if @student.update(student_params)
```

The `require` (here `student`) will fail with a bad request if the required strong param is missing, and all the permitted inner params are optional. Anything else will be filtered out, but again only when this is explicitly called.

See http://edgeguides.rubyonrails.org/action_controller_overview.html#strong-parameters